sqlserver/datadog_checks/sqlserver/sqlserver.py

```diff
@@ -711,6 +712,58 @@ def typed_metric(self, cfg_inst, table, base_name=None, user_type=None, sql_coun
 
         return cls(cfg_inst, base_name, metric_type, column, self.log)
 
+    def _check_connections_by_connecting_to_db(self):
+        for db in self.databases:
+            if db.name != self.connection.DEFAULT_DATABASE:
+                try:
+                    self.connection.check_database_conns(db.name)
+                except Exception as e:
+                    # service_check errors on auto discovered databases should not abort the check
+                    self.log.warning("failed service check for auto discovered database: %s", e)
+
+    def _check_connections_by_use_db(self):
+        with self.connection.open_managed_default_connection():
+            with self.connection.get_managed_cursor() as cursor:
+                for db in self.databases:
+                    check_err_message = "Database {} connection service check failed: {}"
+                    try:
+                        cursor.execute(SWITCH_DB_STATEMENT.format(db.name))
+                    except Exception as e:
+                        self.log.warning(check_err_message.format(db.name, str(e)))
+                        self.handle_service_check(
+                            AgentCheck.CRITICAL,
+                            self.connection.get_host_with_port(),
+                            db.name,
+                            check_err_message.format(db.name, str(e)),
+                            False,
+                        )
+                        continue
+                    try:
+                        cursor.execute(DATABASE_SERVICE_CHECK_QUERY)
+                        cursor.fetchall()
+                    except Exception as e:
+                        self.log.warning(check_err_message.format(db.name, str(e)))
+                        self.handle_service_check(
+                            AgentCheck.CRITICAL,
+                            self.connection.get_host_with_port(),
+                            db.name,
+                            check_err_message.format(db.name, str(e)),
+                            False,
+                        )
+                        continue
+                    self.handle_service_check(AgentCheck.OK, self.connection.get_host_with_port(), db.name, False)
+                #set here back to master
+            with self.connection.get_managed_cursor() as cursor:
```

Comment:
well, I was thinking going back to MASTER DB is like a part of "finally".
So, if we throw before reaching it it's not good.

We can think of making it a part of 'with self.connection.get_managed_cursor()' so that on exit we always switch back to master DB. Or leave it as it is....

Reply:
I think we can reuse the prior cursor instead of creating a new cursor

Reply:
thx, done